src/lnp/java/teammates/lnp/cases/InstructorStudentCascadingUpdateLNPTest.java

```diff
@@ -0,0 +1,282 @@
+    private static final int NUMBER_OF_FEEDBACK_QUESTIONS = 20;
+
+    private static final String INSTRUCTOR_NAME = "LnPInstructor";
+    private static final String INSTRUCTOR_ID = "LnPInstructor_id";
+    private static final String INSTRUCTOR_EMAIL = "tmms.test@gmail.tmt";
+    private static final String COURSE_NAME = "tmms.test.gma-demo";
+    private static final String COURSE_ID = "tmms.test.gma-demo";
+
+    private static final String STUDENT_NAME_PREFIX = "LnPStudent";
+    private static final String STUDENT_ID_PREFIX = "LnPStudent.tmms";
+    private static final String STUDENT_EMAIL_SUBFIX = "@gmail.tmt";
+
+    private static final String FEEDBACK_RESPONSE_PREFIX = "LnPResponse";
+    private static final String FEEDBACK_SESSION_NAME = "LnPSession";
+
+    private static final double ERROR_RATE_LIMIT = 0.01;
+    private static final double MEAN_RESP_TIME_LIMIT = 10;
+
+    @Override
+    protected LNPTestData getTestData() {
+        return new LNPTestData() {
+            @Override
+            protected Map<String, AccountAttributes> generateAccounts() {
+                return new HashMap<>();
+            }
+
+            @Override
+            protected Map<String, CourseAttributes> generateCourses() {
+                Map<String, CourseAttributes> courses = new HashMap<>();
+
+                courses.put(COURSE_ID, CourseAttributes.builder(COURSE_ID)
+                        .withName(COURSE_NAME)
+                        .withTimezone(ZoneId.of("UTC"))
+                        .build()
+                );
+
+                return courses;
+            }
+
+            @Override
+            protected Map<String, StudentAttributes> generateStudents() {
+                Map<String, StudentAttributes> students = new HashMap<>();
+
+                for (int i = 0; i < NUM_STUDENTS; i++) {
+                    students.put(STUDENT_NAME_PREFIX + i, StudentAttributes.builder(COURSE_ID,
+                            STUDENT_NAME_PREFIX + i + STUDENT_EMAIL_SUBFIX)
+                            .withGoogleId(STUDENT_ID_PREFIX + i)
+                            .withName(STUDENT_NAME_PREFIX + i)
+                            .withComment("This student's name is " + STUDENT_NAME_PREFIX + i)
+                            .withSectionName(String.valueOf(i / NUM_STUDENTS_PER_SECTION))
+                            .withTeamName(String.valueOf(i / NUM_STUDENTS_PER_SECTION))
+                            .build());
+                }
+
+                return students;
+            }
+
+            @Override
+            protected Map<String, FeedbackResponseAttributes> generateFeedbackResponses() {
+                Map<String, FeedbackResponseAttributes> feedbackResponses = new HashMap<>();
+
+                for (int i = 1; i <= NUMBER_OF_FEEDBACK_QUESTIONS; i++) {
+                    for (int j = 0; j <= NUM_STUDENTS; j++) {
+                        String responseText = FEEDBACK_RESPONSE_PREFIX + " " + j;
```

Comment:
I wonder if response text should be longer to simulate a reasonable response size, given that the latency of the whole thing would scale with data size

Reply:
hmm I will add more text to it then